### PR TITLE
Remove sampleSpace variables

### PR DESCRIPTION
**Pull request by fzaiser**
By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).

### CODE REVIEW

File: src/Distributions/BernoulliExpNeg/Implementation.dfy [hunk 1]
```diff
@@ -29,8 +29,11 @@ module BernoulliExpNeg.Implementation {
         invariant gamma'.numer >= 0
         invariant Model.GammaReductionLoop((true, gamma))(old(s)) == Model.GammaReductionLoop((b, gamma'))(s)
       {
+        var prevGamma := gamma';
+        var prevS := s;
         b := BernoulliExpNegSampleCaseLe1(Rationals.Int(1));
         gamma' := Rationals.Rational(gamma'.numer - gamma'.denom, gamma'.denom);
+        assert Model.GammaReductionLoop((true, prevGamma))(prevS) == Model.GammaReductionLoop((b, gamma'))(s);
```
Comment (stefan-aws) [inline src/Distributions/BernoulliExpNeg/Implementation.dfy:36]: This seems to be unrelated to the rest of the PR.

**Review by stefan-aws — COMMENTED** [review]
I personally think we should keep the definitions in `Measures.dfy` as generic / close to the established theory as possible, even though the particular event space in `RandomSource` is a trivial one. I like some of the other changes in this PR though.

File: src/Math/Measures.dfy [hunk 2]
```diff
@@ -26,20 +33,12 @@ module Measures {
     f(i) + CountableSum(f, i+1)
   }
 
-  ghost function Powerset<A(!new)>(): iset<A> {
```
Comment (stefan-aws) [inline src/Math/Measures.dfy:29]: Good you remove this one. I don't think the naming was right.

File: src/Math/Measures.dfy [hunk 3]
```diff
@@ -10,10 +10,17 @@ module Measures {
 
   type Probability = x: real | 0.0 <= x <= 1.0
 
-  ghost predicate IsSigmaAlgebra<T(!new)>(eventSpace: iset<iset<T>>, sampleSpace: iset<T>) {
-    && (forall e | e in eventSpace :: e <= sampleSpace)
+  ghost function Complement<T(!new)>(event: iset<T>): iset<T> {
+    iset x: T | x !in event
+  }
+
+  ghost function SampleSpace<T(!new)>(): iset<T> {
```
Comment (stefan-aws) [inline src/Math/Measures.dfy:17]: This function seems very useful, even if we'd keep the generality elsewhere.

File: src/Math/Measures.dfy [hunk 4]
```diff
@@ -10,10 +10,17 @@ module Measures {
...
+    Complement(iset{})
+  }
+
+  ghost predicate IsSigmaAlgebra<T(!new)>(eventSpace: iset<iset<T>>) {
```
Comment (stefan-aws) [inline src/Math/Measures.dfy:21]: I don't think we should remove `sampleSpace` here (and elsewhere). Imo it's confusing for people that read this section for the first time, and it doesn't seem to save us from writing too much code elsewhere.

**Review by stefan-aws — CHANGES_REQUESTED** [review]

File: src/Distributions/BernoulliExpNeg/Implementation.dfy (hunk 1 above)
Comment (fzaiser) [reply src/Distributions/BernoulliExpNeg/Implementation.dfy:36]: The changes I made caused the resource usage of this proof to increase, which is why I had to modify it.

**Comment by stefan-aws** [comment]
Could you add one comment in `Measures.dfy` that the sample space we are working with is "the set of elements of type `T`" (and generally, sample spaces can be obtained by defining appropriate (sub-)types).

**Comment by fzaiser** [comment]
Added some comments around σ-algebras.

**Review by stefan-aws — APPROVED** [review]
